dockerclient.go

```diff
@@ -212,6 +212,50 @@ func (client *DockerClient) ContainerChanges(id string) ([]*ContainerChanges, er
 	return changes, nil
 }
 
+func (client *DockerClient) readJSONStream(stream io.ReadCloser, decode func(*json.Decoder) decodingResult) (<-chan decodingResult, chan<- struct{}) {
```

Comment:
Suggestion: The "pattern" we use in Swarm is to not return the stop channel, but rather pass it as an argument.

The caller is free to pass `nil` if he doesn't wish to ever stop the call. You don't have to check `nil` in your code, go handles nil channel automagically (it won't trigger your `select{}`)